### ISSUE TITLE
Redesign SPI factory to make them into one class

### ISSUE THREAD

**Issue by terrymanu** [apache/shardingsphere-elasticjob#1537]
The current factory using SPI are separate into multiple classes.
It is better redesign and abstract them into one class.

The SPI factory class are:

- JobItemExecutorFactory
- JobShardingStrategyFactory
- JobExecutorServiceHandlerFactory
- JobErrorHandlerFactory
- ElasticJobListenerFactory

**Comment by wwj-go** [comment]
@terrymanu  hi，I am very interested in this task, could you assign it to me ？

**Comment by wwj-go** [comment]
hi @Technoboy- ，I'm sorry to reply so late. In the past few days, I have read the code related to this issues, and have done a lot of thinking and trying. Finally, I found that if you want to solve this problem elegantly, it is best to implement your own SPI loading mechanism to replace java The native SPI mechanism, refer to the practice of [dubbo ](https://github.com/apache/dubbo), I have implemented it, and related issue is [here ](https://github.com/apache/shardingsphere-elasticjob/issues/1546), Later I will submit a new code that implements the SPI mechanism. After the reconstruction of the SPI mechanism is completed, I will continue to complete this problem.
